File: app/src/main/scala/algorithm/lc/TwoSum.scala
package algorithm.lc

import scala.collection.mutable

object TwoSum {

  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    val mm = mutable.Map[Int, Int]()

    nums.zipWithIndex.foreach(p => {
      val (v, j) = (p._1, p._2)
      val k = target - v

      mm.get(k) match {
        case Some(i) =>
          return Array(j, i)
        case None =>
          mm(v) = j
      }
    })

    Array()
  }

  def main(args: Array[String]): Unit = {
    twoSum(Array(1, 2, 3, 5), 5).foreach(i => {
      println(i)
    })
  }
}

File: app/src/main/scala/algorithm/lc/common/TwoSum.java
package algorithm.lc.common;

import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Objects;

public class TwoSum {
    public static List<Integer> twoSum(List<Integer> nums, Integer target) {
        HashMap<Integer, Integer> res = new HashMap<>();

        for (int i = 0; i < nums.size() - 1; i++) {
            Integer v = nums.get(i);
            int k = target - v;
            Integer j = res.get(k);

            if (Objects.nonNull(j)) {
                return Arrays.asList(j, i);
            }

            res.put(v, i);
        }

        return Collections.emptyList();
    }

    public static void main(String[] args) {
        twoSum(Arrays.asList(1, 2, 3, 5), 5).stream()
            .forEach(System.out::println);
    }
}
